src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -21,16 +20,6 @@ object ClusterDnsCache {
 
   @volatile var HostToIp: Map[Host, IP] = Map.empty
```

Comment:
I wonder if we can do away with `HostToIp` and store its contents directly in the Guava cache. We could define something like

```scala
case class DnsCacheValue(host: Host, ip: IP)
```

and then update `CacheLoader.load()` accordingly?

Reply:
I am not sure if this would work, the JupyterNameService talks directly to the ClusterDnsCache object.

Reply:
Ah right. @rtitle, do you know if we have a way to get around that seeming limitation where `JupyterNameService` talks to `ClusterDnsCache` object implicitly via the descriptor. It would be great to not have to have that `var`.

Reply:
I've made [this PR](https://github.com/DataBiosphere/leonardo/pull/627) to this PR's branch to encapsulate the `volatile` and `synchronized` stuff so the client code is cleaner. We could also utilize it for other similar situations in the codebase.